RELEASES.md

```diff
@@ -1,3 +1,440 @@
+and the low-level `ink_storage::lazy::LazyHashMap`. Both provide very similar functionality in that they map some generic key to some storage entity.
+
+However, their APIs look very differently. Whereas the `HashMap` provides a rich and high-level API that is comparable to that of Rust's very own `HashMap`, the `LazyHashMap` provides only a fraction of the API and also operates on `Option<T>` values types instead of `T` directly. It is more similar Solidity mappings than to Rust's `HashMap`.
+
+The fundamental difference of both data structures is that `HashMap` is aware of the keys that have been stored in it and thus can reconstruct exactly which elements and storage regions apply to it. This enables it to provide iteration and automated deletion as well as efficient way to defragment its underlying storage to free some storage space again. This goes very well in the vein of Substrate's storage rent model where contracts have to pay for the storage they are using.
+
+| Data Structure | level of abstraction | caching | lazy | element type | container |
+|:--|:-:|:-:|:-:|:-:|:-:|
+| `T` | - | yes | no | `T` | primitive value |
+| `Lazy<T>` | high-level | yes | yes | `T` | single element container |
+| `LazyCell<T>` | low-level | yes | yes | `Option<T>` | single element, no container |
+| `Vec<T>` | high-level | yes | yes | `T` | Rust vector-like container |
+| `LazyIndexMap<T>` | low-level | yes | yes | `Option<T>` | similar to Solidity mapping |
+| `HashMap<K, V>` | high-level | yes | yes | `V` (key type `K`) | Rust map-like container |
+| `LazyHashMap<K, V>` | low-level | yes | yes | `Option<V>` (key type `K`) | similar to Solidity mapping |
+
+There are many more! For more information about the specifics please take a look into [the `ink_storage` crate documentation](https://paritytech.github.io/ink/ink_storage/).
+
+### Spread & Packed Modes
+
+Storing or loading complex data structures to and from contract storage can be done in many different ways. You could store all information into a single storage cell or you could try to store all information into as many different cells as possible. Both strategies have pros and cons under different conditions.
+
+For example it might be a very good idea to store all the information under the same cell if all the information is very compact. For example when we are dealing with a byte vector that is expected to never be larger than approx a tousand elements it would probably be more efficient if we store all those tousand bytes in the same cell and especially if we often access many of those (or all) in our contract messages.
+
+On the other handside spreading information to as many cells as possible might be much more efficient if we are dealing with big data structures or a lot of information that is not compact or when messages that operate on the data always only need a small fraction of the whole data.
```

Comment:
```suggestion
On the other hand spreading information across as many cells as possible might be much more efficient if we are dealing with big data structures, a lot of information that is not compact, or when messages that operate on the data always only need a small fraction of the whole data.
```
Sentence is a bit long, have added some commas

Reply:
It is good to have a reviewer that knows where to place commas in english sentences. :D